### DISCUSSION TITLE
[2021.08.30] Stack 구현하기

### DISCUSSION THREAD

**Discussion by pyunji** [pyunji/Hyundai-ITnE-Team_QuadCore#77] [Algorithm]
# 아래에 주어진 조건을 통해 스택을 구현하라
### 필드
- `String data`: 현재 노드의 데이터 저장
- `Node<String> next`: 다음 노드의 주소 저장
### 생성자
  - param: `String data`

### 필드
- `Node<String> top` - Stack의 가장 위에 있는 노드
### 메서드
- `String pop()`: Stack의 가장 위에 있는(가장 최근에 들어간) 노드의 데이터를 반환하고 스택에서 제거한다.
  - Stack이 비어있을 경우 `EmptyStackException`을 발생시킨다.
  - `top` 필드 갱신 작업 필요
  - param: 없음
- `void push(String item)`: 스택에 item을 추가한다.
  - param: `String item`
  - `top` 필드 갱신 작업 필요
- `String peek()`: Stack의 가장 위에 있는(가장 최근에 들어간) 노드의 데이터를 반환하고 스택에서 제거하지 않는다.
  - Stack이 비어있을 경우 `EmptyStackException`을 발생시킨다.
  - `top` 필드 갱신 작업 필요 없음
- `boolean isEmpty()`: Stack이 비어있는지 확인한다.

#### ❗ [동영상 참고](https://www.youtube.com/watch?v=whVUYv0Leg0)

**Comment by pyunji** [comment]
[Stack 구현하기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/pyunji/pyunji/DataStructure/Basic_Stack.java)

**Comment by hyk339** [comment]
[Stack 구현하기 solved](https://github.com/hyk339/Hyundai-ITnE-Team_QuadCore/blob/master/khy/dataStructure/Basic_Stack.java)

**Comment by chaeyeon4u** [comment]
[Stack 구현하기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/dataStructure/StackBasic.java)

**Comment by leechang4** [comment]
[Stack 구현하기 solved](https://github.com/leechang4/Hyundai-ITnE-Team_QuadCore/blob/master/lch/DataStructure/Basic_Stack.java)
